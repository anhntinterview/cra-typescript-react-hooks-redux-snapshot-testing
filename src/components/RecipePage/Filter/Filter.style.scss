.filter-wrapper {
    input {
        padding: 0.5rem 1rem;
        border: 1px solid #cdcdcd;
        border-radius: 2rem;
        &:focus {
            border-color: #cdcdcd;
            color: white;
        }
    }
}