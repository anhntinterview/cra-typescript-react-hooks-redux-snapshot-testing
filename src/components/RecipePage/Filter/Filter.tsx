import React, { useState, useCallback, useContext } from "react";
import { debounce } from "lodash";
import { searchRecipes } from "../../../middleware/recipe.middleware";
import { RecipeContext } from "../../../contexts/RecipeContext";
import { ELEMENT_TXT } from "../../../constants/recipe.const";

import "./Filter.style.scss";

export interface FilterProps {}

const Filter: React.FC<FilterProps> = () => {
  const { dispatch } = useContext(RecipeContext);
  const [keyword, setKeyword] = useState<string>("");

  const debounceInput = useCallback(
    debounce((nextValue) => {
      searchRecipes(dispatch, nextValue);
    }, 1000),
    []
  );

  function handleInputOnChange(e: React.ChangeEvent<HTMLInputElement>) {
    const { value } = e.target;
    setKeyword(value);
    debounceInput(value);
  }

  return (
    <div className="filter-wrapper">
      <input
        type="text"
        onChange={handleInputOnChange}
        value={keyword}
        placeholder={ELEMENT_TXT.INPUT_PLACEHOLDER}
      />
    </div>
  );
};

export default Filter;
