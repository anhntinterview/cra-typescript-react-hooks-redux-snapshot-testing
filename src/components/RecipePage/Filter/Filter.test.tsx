import * as React from "react";
import { 
  shallow, 
} from "enzyme";

import Filter from "./Filter";

describe('Filter', () => {
  it('should match a snapshot', () => {
    expect(shallow(<Filter />)).toMatchSnapshot();
  })
})
