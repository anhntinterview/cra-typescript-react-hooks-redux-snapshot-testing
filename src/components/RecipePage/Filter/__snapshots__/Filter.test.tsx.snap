// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should match a snapshot 1`] = `
<div
  className="filter-wrapper"
>
  <input
    onChange={[Function]}
    placeholder="Please typing your keyword ..."
    type="text"
    value=""
  />
</div>
`;
